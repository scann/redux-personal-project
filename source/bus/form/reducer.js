import { combineForms } from 'react-redux-form';

const initialFormState = {
    newTaskMessage: '',
};

export const formReducer = combineForms({
    scheduler: {
        initialFormState,
    },
}, 'form');
