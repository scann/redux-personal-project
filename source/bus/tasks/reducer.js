//Core
import { fromJS, List } from 'immutable';

//Instruments
import { types } from './types';

const initialState = List();

export const tasksReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.FILL_TASKS:
            return fromJS(action.payload);

        case types.CREATE_TASK:
            return state.unshift(fromJS(action.payload));

        case types.REMOVE_TASK:
            const taskId = fromJS(action.payload);

            return state.filter((task) => task.get('id') !== taskId);

        case types.COMPLETE_ALL_TASKS:

            return state.map((task) => task.set('completed', true));

        case types.UPDATE_TASK:
            return state.update(
                state.findIndex(
                    (task) => task.get('id') === action.payload.id),
                () => fromJS(action.payload)
            );

        default:
            return state;
    }
};
