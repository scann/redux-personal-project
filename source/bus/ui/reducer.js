//Core
import { Map } from 'immutable';

//Types
import { types } from './types';

const initialState = Map({
    isTasksFetching: false,
});

export const uiReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.START_SPINNING:
            return state.set('isTasksFetching', true);

        case types.STOP_SPINNING:
            return state.set('isTasksFetching', false);

        default:
            return state;
    }
};
