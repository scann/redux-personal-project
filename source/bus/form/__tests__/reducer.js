//Reducer
import { formReducer } from '../reducer';

describe('form reducer', () => {
    test('should return initial state by default', () => {
        expect(formReducer(void 0, {})).toMatchSnapshot();
    });
});
