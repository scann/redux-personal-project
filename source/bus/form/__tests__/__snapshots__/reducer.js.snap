// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form reducer should return initial state by default 1`] = `
Object {
  "forms": Object {
    "$form": Object {
      "errors": Object {},
      "focus": false,
      "initialValue": Object {
        "scheduler": Object {
          "initialFormState": Object {
            "newTaskMessage": "",
          },
        },
      },
      "intents": Array [],
      "model": "form",
      "pending": false,
      "pristine": true,
      "retouched": false,
      "submitFailed": false,
      "submitted": false,
      "touched": false,
      "valid": true,
      "validated": false,
      "validating": false,
      "validity": Object {},
      "value": Object {
        "scheduler": Object {
          "initialFormState": Object {
            "newTaskMessage": "",
          },
        },
      },
    },
    "scheduler": Object {
      "$form": Object {
        "errors": Object {},
        "focus": false,
        "initialValue": Object {
          "initialFormState": Object {
            "newTaskMessage": "",
          },
        },
        "intents": Array [],
        "model": "form.scheduler",
        "pending": false,
        "pristine": true,
        "retouched": false,
        "submitFailed": false,
        "submitted": false,
        "touched": false,
        "valid": true,
        "validated": false,
        "validating": false,
        "validity": Object {},
        "value": Object {
          "initialFormState": Object {
            "newTaskMessage": "",
          },
        },
      },
      "initialFormState": Object {
        "$form": Object {
          "errors": Object {},
          "focus": false,
          "initialValue": Object {
            "newTaskMessage": "",
          },
          "intents": Array [],
          "model": "form.scheduler.initialFormState",
          "pending": false,
          "pristine": true,
          "retouched": false,
          "submitFailed": false,
          "submitted": false,
          "touched": false,
          "valid": true,
          "validated": false,
          "validating": false,
          "validity": Object {},
          "value": Object {
            "newTaskMessage": "",
          },
        },
        "newTaskMessage": Object {
          "errors": Object {},
          "focus": false,
          "initialValue": "",
          "intents": Array [],
          "model": "form.scheduler.initialFormState.newTaskMessage",
          "pending": false,
          "pristine": true,
          "retouched": false,
          "submitFailed": false,
          "submitted": false,
          "touched": false,
          "valid": true,
          "validated": false,
          "validating": false,
          "validity": Object {},
          "value": "",
        },
      },
    },
  },
  "scheduler": Object {
    "initialFormState": Object {
      "newTaskMessage": "",
    },
  },
}
`;
